src/main/kotlin/no/nav/familie/ef/sak/vedtak/uttrekk/UttrekkArbeidssøkerService.kt

```diff
@@ -0,0 +1,64 @@
+package no.nav.familie.ef.sak.vedtak.uttrekk
+
+import no.nav.familie.ef.sak.infrastruktur.sikkerhet.TilgangService
+import no.nav.familie.ef.sak.repository.findByIdOrThrow
+import no.nav.familie.ef.sak.vedtak.domain.AktivitetType
+import no.nav.familie.ef.sak.vedtak.domain.Vedtaksperiode
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+import java.time.YearMonth
+import java.util.UUID
+
+@Service
+class UttrekkArbeidssøkerService(
+        private val tilgangService: TilgangService,
+        private val uttrekkArbeidssøkerRepository: UttrekkArbeidssøkerRepository,
+) {
+
+    fun forrigeMåned(): () -> YearMonth = { YearMonth.now().minusMonths(1) }
+
+    @Transactional
+    fun opprettUttrekkArbeidssøkere(årMåned: YearMonth = forrigeMåned().invoke()) {
+        hentArbeidssøkere(årMåned).forEach {
+            uttrekkArbeidssøkerRepository.insert(UttrekkArbeidssøkere(fagsakId = it.fagsakId,
+                                                                      vedtakId = it.behandlingIdForVedtak,
+                                                                      årMåned = årMåned))
+        }
+    }
+
+    fun settKontrollert(id: UUID, kontrollert: Boolean) {
+        val uttrekkArbeidssøkere = uttrekkArbeidssøkerRepository.findByIdOrThrow(id)
+        tilgangService.validerTilgangTilFagsak(uttrekkArbeidssøkere.fagsakId)
+        uttrekkArbeidssøkerRepository.update(uttrekkArbeidssøkere.copy(kontrollert = kontrollert))
+    }
+
+    fun hentUttrekkArbeidssøkere(årMåned: YearMonth = forrigeMåned().invoke()): UttrekkArbeidssøkereDto {
+        val arbeidssøkere = uttrekkArbeidssøkerRepository.findAllByÅrMåned(årMåned)
+        return UttrekkArbeidssøkereDto(årMåned = årMåned,
+                                       antallTotalt = arbeidssøkere.size,
+                                       antallKontrollert = arbeidssøkere.count { it.kontrollert ?: false },
+                                       arbeidssøkere = arbeidssøkere.map(UttrekkArbeidssøkere::tilDto))
+    }
+
+    fun hentArbeidssøkere(årMåned: YearMonth = forrigeMåned().invoke()): List<ArbeidsssøkereTilUttrekk> {
+        val startdato = årMåned.atDay(1)
+        val sluttdato = årMåned.atEndOfMonth()
+        val arbeidssøkere = uttrekkArbeidssøkerRepository.hentArbeidssøkere(startdato, sluttdato)
+        return arbeidssøkere.filter { harPeriodeSomArbeidssøker(it, startdato, sluttdato) }
+    }
+
+    private fun harPeriodeSomArbeidssøker(it: ArbeidsssøkereTilUttrekk,
+                                          startdato: LocalDate,
+                                          sluttdato: LocalDate) =
+            it.perioder.perioder.any {
+                it.datoFra <= startdato
+                && it.datoTil >= sluttdato
```

Comment:
hvorfor må du gjøre filtrering på datoer her? Skal ikke dette gjelde for personer som får beløp for en halv måned etc?

Reply:
Når vi henter vedtaket så inneholder den en liste med perioder.
Vi skal sjekke att det finnes en period i forrige måned som er av typen arbeidssøker.

Reply:
Tenkte på at det filtreres på perioder i sql'n og at man ikke trengte det her, men dette sjekker kanskje noe annet

Reply:
Ja, god observert 👍 
Grunnen til att det er gjort sånn her er att queryn filtrerer ut de med perioder en gitt måned, men får ikke filtrert ut de som har typen arbeidssøker og en gitt periode på en god måte pga dataen som ligger i json.. Eller jeg får gjort det, men det blir en relativt stygg query :P